R/pkg/R/mllib_regression.R

```diff
@@ -159,10 +161,16 @@ setMethod("spark.glm", signature(data = "SparkDataFrame", formula = "formula"),
               weightCol <- as.character(weightCol)
             }
 
+            if (!is.null(offsetCol) && offsetCol == "") {
+              offsetCol <- NULL
+            } else if (!is.null(offsetCol)) {
+              offsetCol <- as.character(offsetCol)
+            }
```

Comment:
perhaps
```
if (!is.null(offsetCol)) {
  offsetCol <- as.character(offsetCol)
  if (nchar(offsetCol) == 0) {
    offsetCol <- NULL
  }
}
```

not sure if you want to cover other cases when offsetCol cannot be coerced - eg. NA